Assets/Python/EntryPoints/CvRandomEventInterface.py

```diff
@@ -6343,6 +6343,24 @@ def getHelpVolcanoExtinction(argsList):
 
 ### Fire events for C2C
 
+def doWildFire(argsList):
+	data = argsList[1]
+	CyPlayer = GC.getPlayer(data.ePlayer)
+	CyCity = CyPlayer.getCity(data.iCityId)
+
+	for i in range(GC.getNumBuildingInfos()):
+		if isLimitedWonder(i) or not CyCity.hasBuilding(i) or CyCity.isFreeBuilding(i):
+			continue
+		info = GC.getBuildingInfo(i)
+		if info.getProductionCost() < 1 or info.isNukeImmune() or info.isAutoBuild():
+			continue
+	        iBurnBuilding = i
+
+	if iBurnBuilding != -1:
```

Comment:
iBurnBuilding needs to be set to be -1 before the for loop so it won't fritz out for cities with no valid buildings to burn.

Reply:
Another thing here that should be improved is to add all the valid buildings to a list, and then pick a random one of them from the list to burn. Right now it will always burn the building with the highest index value in the city which will mean buildings will burn down in a specific order, that lower index buildings will be very well insulated and rarely burn because of it.

Reply:
> Another thing here that should be improved is to add all the valid buildings to a list, and then pick a random one of them from the list to burn. Right now it will always burn the building with the highest index value in the city which will mean buildings will burn down in a specific order, that lower index buildings will be very well insulated and rarely burn because of it.

An example of what I mean:
```py
def applyMercenariesAncient2(argsList):
	data = argsList[1]
	listPlots = []
	MAP = GC.getMap()
	for i in xrange(MAP.numPlots()):
		plot = MAP.plotByIndex(i)
		if (plot.getOwner() == -1 and not plot.isWater() and not plot.isImpassable()
		and plot.area().getCitiesPerPlayer(data.ePlayer) > 0
		and plot.isAdjacentPlayer(data.ePlayer, True)
		):
			listPlots.append(i)

	if not listPlots:
		return

	plot = MAP.plotByIndex(listPlots[GAME.getSorenRandNum(len(listPlots), "Mercenary event location")])
```

Reply:
hmm very interesting thing i noticed as well in other burning events is that it always burns the same houses. I guess this was the cause.